### PR TITLE
[SofaKernel] issofa_springff

### PR DESCRIPTION
**Pull request by untereiner**
### Improvements
- SpringForceField: enabled flag in SpringForceField::Spring 
- SpringForceField: make all Data public to be able to edit the springs info from other components 
- SpringForceField: fix duplicated member 

### Moved files 

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: SofaKernel/modules/SofaDeformable/SpringForceField.inl [hunk 1]
```diff
@@ -126,7 +126,7 @@ void SpringForceField<DataTypes>::addSpringForce(Real& ener, VecDeriv& f1, const
     int b = spring.m2;
     Coord u = p2[b]-p1[a];
     Real d = u.norm();
-    if( d<1.0e-4 ) // null length => no force
+    if( spring.enabled && d<1.0e-4 ) // null length => no force
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaDeformable/SpringForceField.inl]: Shouldn't it be !spring.enabled ?

Comment (damienmarchal) [reply SofaKernel/modules/SofaDeformable/SpringForceField.inl]: Sorry the code is right.

**Comment by damienmarchal** [comment]
Hi all, 

Sadely it seems this PR have broken the windows build: https://www.sofa-framework.org/dash/
Probably a link _declspec_ problem as usual ? 

DM. 

EDIT: about declspec...maybe not...

**Comment by guparan** [comment]
Not sure, it may be a cache problem... I launch a fresh build.

**Comment by damienmarchal** [comment]
ok

**Comment by damienmarchal** [comment]
Would it be possible to merge PR #471 in // ? I would like to have PR #410 cleaned/rebased for review so that we can progress on issue #517 :)

**Comment by guparan** [comment]
Done ;-)

**Pull request merged**